### PR TITLE
Proposed adjustments to the project lifecycle

### PR DESCRIPTION
**Pull request by jmertic**
Signed-off-by: John Mertic <jmertic@linuxfoundation.org>

### CODE REVIEW

File: process/lifecycle.md [hunk 1]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
 
 ASWF may adopt or amend this document by votes of its Technical Advisory Council (“TAC”) and Governing Board.
 
-This document provides for three lifecycle stages for contributed projects:
+# Proposal Process
 
+Projects must be submitted to [tac@lists.aswf.io](mailto:tac@lists.aswf.io) for consideration. Project proposals submitted to the ASWF TAC must provide the following information to the best of thier ability.
```
Comment (jfpanisset) [inline process/lifecycle.md:13]: thier -> their

File: process/lifecycle.md [hunk 2]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+
+* Name of the project (existing or proposed)
+* Requested project maturity level (select one): Adopted, Incubation, and Sandbox
+* Project description (please describe the purpose and function of the project, its origin and its significance to the ecosystem)
+* Please explain how this project is aligned with the mission of ASWF?
```
Comment (jfpanisset) [inline process/lifecycle.md:18]: why a question mark?

File: process/lifecycle.md [hunk 3]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+* What is the project’s license for code contributions and methodology for code contributions. ASWF maintains recommendations for contribution and licensing for hosted projects.
+* What tool or platform is utilized for source control (GitHub, etc.) and what is the location (e.g. URL)?
+* What are the external dependencies of the project, and what are the licenses of those dependencies?
+* What roles does the project have (e.g. maintainers, committers?) Who are the current core committers of the project, or which can a list of committers be found?
```
Comment (jfpanisset) [inline process/lifecycle.md:22]: which -> where

File: process/lifecycle.md [hunk 4]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+
+# TAC presentation
+
+Projects will present at the next TAC meeting no sooner than two weeks after their submission date.
+
+Proposed projects will receive a 30-45 minute presentation slot at an upcoming TAC meeting as the schedule allows. The project proposal must include a presentation that conforms to the presentation structure below to ensure uniform and complete submissions.
+
+*   Overview of the project and its purpose
+    *  If a project is a new project, it’s often helpful to share the vision for the project, the anticipated structure and benefits.
+    *  If the project is an existing open source project proposed to join ASWF, it is often helpful to schedule a project walkthrough for the community to better understand the project, including the architecture, structure and how to get starting using the project. This often helps progress reviews more efficiently.
```
Comment (jfpanisset) [inline process/lifecycle.md:39]: starting -> started
helps progress reviews -> helps reviews progress ?

File: process/lifecycle.md [hunk 5]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+*   How does this submission support the ASWF Mission and Vision Statements?
+*   Does the solution have any users?
```
Comment (jfpanisset) [inline process/lifecycle.md:41]: solution -> project ?

File: process/lifecycle.md [hunk 6]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+*   Does the solution have any users? 
+    *   How do you plan to attract users if accepted?
+*   How many committers are you planning to have upfront, and from which companies? 
+    *   How do you plan to attract committers and contributors if accepted?
+*   Architecture (functional / non-functional aspects), design, feature overview
+*   Demo and/or walk through of the project
+
+An initial review of a proposal submitted to the TAC should be conducted within two to four weeks following acceptance of a project contribution proposal. 
+
+The TAC may consider the project for approval at the proposed stage during the meeting it is presented at, provided there is a quorum of TAC voting members present. If there is not a quorum of TAC voting members present or if the TAC is not ready to consider the project for approval during the meeting, the TAC may consider discussing and/or voting on the project proposal via email or postpone the vote to a future meeting. An affirmative vote majority vote of the TAC is required to accept a new TAC Project.
```
Comment (jfpanisset) [inline process/lifecycle.md:50]: affirmative vote majority vote -> affirmative majority vote ?

File: process/lifecycle.md [hunk 7]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+
+* Early-stage projects that the ASWF TAC believes warrant experimentation.
+* New projects that are designed to extend one or more TAC projects with functionality or interoperability libraries.
+* Independent projects that fit the ASWF mission/vision and provide the potential for a novel approach to existing functional areas (or are an attempt to meet an unfulfilled need).
+* Projects commissioned or sanctioned by the ASWF, including initial code for ASWF Working Group collaborations, and "experimental" projects.
+* Any project that realistically intends to join ASWF Incubation in the future and wishes to lay the foundations for that.
+
+### Requirements
 
 To be accepted at the Incubation stage, a project must:
```
Comment (jfpanisset) [inline process/lifecycle.md:75]: Incubation stage -> Sandbox stage

File: process/lifecycle.md [hunk 8]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
 
-An initial review of a proposal submitted to the TAC should be conducted within two to four weeks following acceptance of a project contribution proposal. If a project is a new project, it’s often helpful to share the vision for the project, the anticipated structure and benefits.
-If the project is an existing open source project proposed to join ASWF, it is often helpful to schedule a project walkthrough for the community to better understand the project, including the architecture, structure and how to get starting using the project. This often helps progress reviews more efficiently.
+### Benefits
 
-## Graduation Review
+The Sandbox Stage benefits are outlined below.
 
-To be advance to an Adopted stage, a project must meet the Incubation stage requirements plus:
+* Neutral hosting of the project’s community and any key assets (e.g. trademark, domain, etc.).
```
Comment (jfpanisset) [inline process/lifecycle.md:87]: Previous language about Incubation Stage requirements had a line about transferring trademarks, domain... which got removed for Sandbox Stage, but this talks about "neutral hosting of trademarks and domains". Is this indeed the right stage to do this?

File: process/lifecycle.md [hunk 9]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
-# Project Benefits Associated with Each Lifecycle Stage
+It’s expected that projects in the Sandbox Stage move to the Incubation Stage within one year. In the case of a Sandbox Stage project that is not renewed with ASWF, the trademark and any other assets will be returned to the project maintainers or an organization they designate.
+
+## Incubation Stage Requirements
+
+A project at the Incubation Stage has begun to form a community and develop its scope and mission. Incubation Stage projects likely will have some adoption in testing or production use.
+
+### Requirements
+
+To be accepted at the Incubation stage, a project must meeting the Sandbox requirements plus:
```
Comment (jfpanisset) [inline process/lifecycle.md:105]: meeting -> meet

File: process/lifecycle.md [hunk 10]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+*   Have defined its technical governance, including:
+    *   A LICENSE file in every code repository, with the license chosen an OSI-approved license.
+    *   A README file welcoming new community members to the project and explaining why the project is useful and how to get started.
+    *   A CONTRIBUTING file explaining to other developers and your community of users how to contribute to the project. The file should explain what types of contributions are needed and how the process works.
+    *   A CODEOWNERS or COMMITTERS file to define individuals or teams that are responsible for code in a repository; document current project owners and current and emeritus committers. 
+    *   A CODE_OF_CONDUCT file that sets the ground rules for participants’ behavior associated and helps to facilitate a friendly, welcoming environment. By default, projects should leverage the Linux Foundation Code of Conduct unless an alternate Code of Conduct is approved prior.
```
Comment (jfpanisset) [inline process/lifecycle.md:113]: is approved prior -> prior to what? Maybe "was previously approved"?

File: process/lifecycle.md [hunk 11]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+    *   A RELEASE file that provides documentation on the release methodology, cadence, criteria, etc.
+    *   A GOVERNANCE file that documents the project’s technical governance.
+    *   A SUPPORT file to let users and developers know about ways to get help with your project.
+*   Have achieved and maintained a Core Infrastructure Initiative [Best Practices Badge](https://bestpractices.coreinfrastructure.org/).
```
Comment (jfpanisset) [inline process/lifecycle.md:117]: Should this specify the CII level required (Passing / Silver / Gold)?

File: process/lifecycle.md [hunk 12]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
 * Neutral hosting of the project’s community and any key assets (e.g. trademark, domain, etc.)
 * Assistance from the ASWF TAC to facilitate collaboration with other project communities.
 * Blog announcement or similar communication announcing the inclusion of the project.
 * Right to refer to the project as an incubation project of ASWF, and an opportunity to participate in events and other collaborative activities sponsored by ASWF.
 * Subject to applicable trademark usage guidelines, to display ASWF’s logo on the project’s code repository.
 
-Incubation stage projects are expected to leverage third-party public code repositories.
+### Expectations
+
+Incubation Stage projects should provide a quarterly report to the TAC outlining its progress on completing the requirements for the Active Stage.
```
Comment (jfpanisset) [inline process/lifecycle.md:146]: Active Stage -> Adopted Stage ?

File: process/lifecycle.md [hunk 13]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
 * Blog announcement or similar communication announcing the inclusion of the project.
 * Right to refer to the project as an incubation project of ASWF, and an opportunity to participate in events and other collaborative activities sponsored by ASWF.
 * Subject to applicable trademark usage guidelines, to display ASWF’s logo on the project’s code repository.
 
-Incubation stage projects are expected to leverage third-party public code repositories.
+### Expectations
+
+Incubation Stage projects should provide a quarterly report to the TAC outlining its progress on completing the requirements for the Active Stage. 
+
+Every 12 months, each Incubation Stage project will be reviewed by the TAC to assess its progress towards graduating to the Active Stage. If the project has not met the requirements for graduating to the Active Stage, the TAC may renew the project at the Incubation Stage for another 12 months with a majority vote of the TAC.
```
Comment (jfpanisset) [inline process/lifecycle.md:148]: Active Stage -> Adopted Stage? (2x)

File: process/lifecycle.md [hunk 14]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+
+Adopted Stage projects are considered mature projects that generally are ready for production use. Projects at this stage are focused on growing an ecosystem of users and are often being leveraged in vendor products or being used by end-users.
+
+### Requirements
+
+To be advance to an Adopted stage, a project must meet the Incubation stage requirements plus:
+
+* Demonstrate a substantial ongoing flow of commits and merged contributions, authored by a healthy number of diverse contributors*.
+* Demonstrable roadmap progress.
+* A healthy number of public adopters, identified within the project ( using an ADOPTERS.md page or having showcased on the project’s website )
```
Comment (jfpanisset) [inline process/lifecycle.md:162]: Trivial detail, but in previous sections we didn't specify that project files had to be in Markdown with a .md extension. Also this bullet item list has a mix of period and non period terminated items

File: process/lifecycle.md [hunk 15]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+* Have achieved and maintained a Core Infrastructure Initiative [Best Practices Gold Level Badge](https://bestpractices.coreinfrastructure.org/)
```
Comment (jfpanisset) [inline process/lifecycle.md:163]: That's a pretty strong requirement we haven't made before, if I remember well "Gold" was more an aspirational goal than a hard requirement. Definitely worth discussing, I don't think any of our existing Adopted Stage projects meet it for now (are these Adopted Stage requirements retroactive?)

**Review by jfpanisset — COMMENTED** [review]
I believe this should be discussed at the next TAC meeting since it represents fairly significant policy changes.

**Comment by jmertic** [comment]
> I believe this should be discussed at the next TAC meeting since it represents fairly significant policy changes.

Hey @jfpanisset - yes the plan is to present these proposed changes and the motivations at the next TAC meeting. I left this as draft while I work with @kdt3rd on this.

File: process/lifecycle.md (hunk 8 above)
Comment (jmertic) [reply process/lifecycle.md:87]: Yes - any new assets will be neutrally hosted anywhere in this lifecycle. Existing ones should transition as the project reaches the Incubation stage. Let me know if this answers your question

File: process/lifecycle.md (hunk 15 above)
Comment (jmertic) [reply process/lifecycle.md:163]: Makes sense 👍

File: process/lifecycle.md [hunk 16]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+* Have a technical lead appointed for representation of the project to the TAC.
```
[the remark this answers is not in the document]
Comment (jmertic) [reply process/lifecycle.md:164]: Another good discussion point. For context, I've seen other similar umbrellas move this up to the top level as well, as typically projects in earlier stages will be more focused on progressing through the lifecycle and having TAC voting responsibility can be a distraction.

File: process/lifecycle.md [hunk 17]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+* Be deemed by the TAC to add value to the mission of ASWF.
+* Obtain both a ⅔ supermajority vote of the TAC and an affirmative majority vote of the Governing Board
```
[the remark this answers is not in the document]
Comment (jmertic) [reply process/lifecycle.md:166]: Agreed on both points.

File: process/lifecycle.md (hunk 8 above)
Comment (lgritz) [reply process/lifecycle.md:87]: Seems like a lot of the things that need a sandbox are so preliminary that they probably have not accumulated any property. If the project exists enough to have web sites, trademarks, or even a pile of code, it's probably ready for incubation.

File: process/lifecycle.md (hunk 15 above)
Comment (lgritz) [reply process/lifecycle.md:163]: This deserves a careful look. Even for the existing badge, a lot of projects got hung up on how exactly to satisfy badge requirements that didn't map well to the project at all (encryption, hosting security, password management, that sort of thing). I would need to read the gold requirements really carefully, but I could easily imagine that gold has even more requirements that are inapplicable and will make people spend a lot of time fretting over them. Something to watch out for.

File: process/lifecycle.md [hunk 18]
```diff
@@ -9,69 +8,205 @@ This lifecycle document is maintained by The Academy Software Foundation (“ASW
...
+
+*Since these metrics can vary significantly depending on the type, scope and size of a project, the TAC has final judgment over the level of activity that is adequate to meet these criteria. In general it will be necessary to (at least) demonstrate that the project is not overly reliant on any one individual or company, and can remain healthy in the event of a departure.
```
[the remark this answers is not in the document]
Comment (lgritz) [reply process/lifecycle.md:168]: Maybe this escape hatch is enough to alleviate the fears of my earlier comment.

**Comment by lgritz** [comment]
LGTM overall. I really like the idea of sandbox projects.

File: process/lifecycle.md (hunk 15 above)
Comment (bcipriano) [reply process/lifecycle.md:163]: +1

**Comment by jmertic** [comment]
I've broken this out into...

#267 
#268
#269
#270

Will close this PR and manage through those 4.

**Pull request closed without merging**